src/core/index.ts

```diff
@@ -19,9 +29,165 @@ export class DevAssist {
 
 	private async startTask(task?: string): Promise<void> {
 		this.messages = [];
-		await this.providerRef.deref()?.postStateToWebview();
-
+		this.apiConversationHistory = [];
+		// TODO: At each prominent stage we need to show a message to the user as a update.
+		// await this.providerRef.deref()?.postStateToWebview(); // This should be called after each message update and handled in UI.
+		
 		await this.say("text", task);
+
+		await this.initiateTaskLoop([
+			{
+				type: "text",
+				text: `<task>\n${task}\n</task>`,
+			}
+		]);
+	}
+
+	private async initiateTaskLoop(userContent: any): Promise<void> {
+		let nextUserContent = userContent;
+		let includeFileDetails = true;
+		const didEndLoop = await this.recursivelyMakeClaudeRequests(nextUserContent, includeFileDetails);
+		includeFileDetails = false; // we only need file details the first time
+	}
+	
+
+	async recursivelyMakeClaudeRequests(
+		userContent: any,
+		includeFileDetails: boolean = false
+	): Promise<boolean> {
+		if (this.abort) {
+			throw new Error("DevAssist instance aborted")
+		}
+
+		await this.say(
+			"api_req_started",
+			JSON.stringify({
+				request:userContent
+			})
+		);
+
+		// TODO: FIXME: Environment is coming Undefined
+		// const environmentDetails = await this.loadContext(userContent, includeFileDetails)
+		const environmentDetails = "NONE"
+		// add environment details as its own text block, separate from tool results
+		userContent.push({ type: "text", text: environmentDetails })
+
+		await this.addToApiConversationHistory({ role: "user", content: userContent })
+
+		try {
+			
+			this.assistantMessageContent = []
+			this.didCompleteReadingStream = false
+			this.userMessageContent = []
+
+			const stream = this.attemptApiRequest(-1) // TODO -1 is a placeholder for now. For multiple communication we need to replace it by last message index
+			let assistantMessage = ""
+			try {
+				for await (const chunk of stream) {
+					assistantMessage += chunk.text
+					// TODO: parse raw assistant message into content blocks
+					// TODO: present content to user
+					console.log(assistantMessage);
+				}
+			} catch (error) {
+				this.abortTask() 
+			}
+
+			this.didCompleteReadingStream = true
+
+			
+			
+			let didEndLoop = false
+			if (assistantMessage.length > 0) {
+				await this.addToApiConversationHistory({
+					role: "assistant",
+					content: [{ type: "text", text: assistantMessage }],
+				});
+
+				console.log(this.assistantMessageContent);
+				// TODO: Need to identify in the response any of the tool used or not. 
+				// const didToolUse = this.assistantMessageContent.some((block) => block.type === "tool_use")
+				const didToolUse = false;
```

Comment:
Can you confirm if we are not handling tool_use in the current implementation. Are we also planning on sending the the API conversation to the Webview-ui or just the response?

Reply:
The current implementation includes tool usage functionalities, such as execute_cmd and write_to_file. However, these operations are currently performed without interacting with the user via the UI for confirmation.

Moving forward, I plan to display the success message to the user. This message will be extracted from the final part of the response, wrapped inside the <attempt_completion> tag.